src/main/java/org/folio/innreach/domain/service/impl/MARCTransformationOptionsSettingsServiceImpl.java

```diff
@@ -0,0 +1,114 @@
+    repository.findOneByCentralServerId(centralServerId).ifPresent(m -> {
+      throw new EntityExistsException("MARC Transformation Options Settings for centralServerId = " + centralServerId
+        + "already exist.");
+    });
+
+    var marcTransformOptSet = mapper.toEntity(marcTransformOptSetDTO);
+    marcTransformOptSet.getModifiedFieldsForContributedRecords().forEach(modifiedField -> modifiedField.setMARCTransformationOptionsSettings(marcTransformOptSet));
+    marcTransformOptSet.setCentralServer(centralServerRef(centralServerId));
+    var createdMARCTransformOptSet = repository.save(marcTransformOptSet);
+    return mapper.toDto(createdMARCTransformOptSet);
+  }
+
+  @Override
+  public MARCTransformationOptionsSettingsDTO update(UUID centralServerId, MARCTransformationOptionsSettingsDTO marcTransformOptSetDTO) {
+    var marcTransformOptSet = repository.findOneByCentralServerId(centralServerId).orElseThrow(
+      () -> new EntityNotFoundException(TEXT_MARC_TRANSFORM_OPT_SET_NOT_FOUND + centralServerId));
+
+    var updated = mapper.toEntity(marcTransformOptSetDTO);
+    updateMARCTransformOptSet(marcTransformOptSet, updated);
+    updateModifiedFields(marcTransformOptSet, updated);
+
+    repository.save(marcTransformOptSet);
+
+    return mapper.toDto(marcTransformOptSet);
+  }
+
+  @Override
+  @Transactional
+  public void delete(UUID centralServerId) {
+    var marcTransformOptSet = repository.findOneByCentralServerId(centralServerId).orElseThrow(
+      () -> new EntityNotFoundException(TEXT_MARC_TRANSFORM_OPT_SET_NOT_FOUND + centralServerId));
+    repository.delete(marcTransformOptSet);
+  }
+
+  @Override
+  public MARCTransformationOptionsSettingsListDTO getAll(int offset, int limit) {
+    var marcTransformOptSetList = collectMARCTransformOptSet(offset, limit);
+
+    var marcTransformOptSetListDTO = new MARCTransformationOptionsSettingsListDTO();
+    marcTransformOptSetListDTO.setMaRCTransformOptSetList(marcTransformOptSetList);
+    marcTransformOptSetListDTO.setTotalRecords(marcTransformOptSetList.size());
+
+    return marcTransformOptSetListDTO;
+  }
+
+  private List<MARCTransformationOptionsSettingsDTO> collectMARCTransformOptSet(Integer offset, Integer limit) {
+    return repository.findAll(PageRequest.of(offset, limit))
+      .stream()
+      .map(mapper::toDto)
+      .collect(Collectors.toList());
+  }
+
+  private void updateMARCTransformOptSet(MARCTransformationOptionsSettings marcTransformOptSet, MARCTransformationOptionsSettings updated) {
+    marcTransformOptSet.setExcludedMARCFields(updated.getExcludedMARCFields());
+    marcTransformOptSet.setConfigIsActive(updated.getConfigIsActive());
+  }
+
+  private void updateModifiedFields(MARCTransformationOptionsSettings marcTransformOptSet, MARCTransformationOptionsSettings updated) {
+    var currentModifiedFields = new ArrayList<>(marcTransformOptSet.getModifiedFieldsForContributedRecords());
```

Comment:
Note that a new list is also created in `merge` method, no need to create new list here